### ISSUE TITLE
use_build_context_synchronously and try/catch

### ISSUE THREAD

**Issue by slavap** [dart-lang/linter#3466]
No warning in that case, though it should be:
```dart
try {
  await aaa(); // SUCCESS
  await bbb(); // FAILURE
} catch (ex, stack) {
  await ccc(context);
}
```

**Comment by srawlins** [comment]
CC @goderbauer

**Comment by srawlins** [comment]
Oops I think a duplicate of https://github.com/dart-lang/linter/issues/3028

**Comment by pq** [comment]
Yep.  This is a dup.  Closing in favor of #3028.
